services/workflows-service/src/workflow/workflow.controller.external.ts

```diff
@@ -293,4 +298,54 @@ export class WorkflowControllerExternal {
 
     return;
   }
+
+  @common.Post('/with-token')
+  @swagger.ApiOkResponse()
+  @common.HttpCode(200)
+  @swagger.ApiForbiddenResponse({ type: errors.ForbiddenException })
+  @UseCustomerAuthGuard()
+  async createWorkflowRuntimeDataWithToken(
+    @common.Body() body: WorkflowWithToken,
+    @CurrentProject() currentProjectId: TProjectId,
+  ) {
+    const endUser = await this.endUserService.createWithBusiness(
+      {
+        firstName: '',
+        lastName: '',
+        email: body.email,
+        companyName: '',
+      },
+      currentProjectId,
+    );
+
+    const workflow = await this.service.createOrUpdateWorkflowRuntime({
+      workflowDefinitionId: body.workflowDefinitionId,
+      context: {
+        entity: {
+          ballerineEntityId: endUser.businesses.at(-1)?.id,
+          type: 'business',
+          data: {
+            additionalInformation: {
+              endUserId: endUser.id,
+            },
+          },
+        },
+        documents: [],
+      },
+      projectIds: [currentProjectId],
+      currentProjectId: currentProjectId,
+    });
+
+    const nowPlus30Days = new Date(Date.now() + 30 * 24 * 60 * 60 * 1000);
+
+    const workflowToken = await this.workflowTokenService.create(currentProjectId, {
+      workflowRuntimeDataId: workflow[0].workflowRuntimeData.id,
+      endUserId: endUser.id,
+      expiresAt: nowPlus30Days,
+    });
+
+    // @TODO: Send email with token
+
+    return { token: workflowToken.token };
+  }
 }
```

Comment:
This endpoint is temporary. Daniel suggested that the implementation should be more generic, leveraging our workflow infra. The token generation and sending the email should be part of a workflow. But it requires new plugins and new workflow definition, I wanted to implement it ASAP.